odoo/models.py

```diff
@@ -5991,6 +5992,18 @@ def process(res):
                 ))
 
         if onchange in ("1", "true"):
+            # before calling on change ensure float are properly rounded
+            # as expected in configuration
+            field = self._fields[field_name]
+            if field.type == 'float' and field._digits:
+                setattr(
+                    self,
+                    field_name,
+                    float_round(
+                        getattr(self, field_name),
+                        precision_digits=self.env['decimal.precision'].precision_get(field._digits)
```

Comment:
Tests remind me that `_digits` can be a string or a tuple as [documented](https://www.odoo.com/documentation/14.0/fr/developer/reference/addons/orm.html#odoo.fields.Float) I've to manage that properly !